### PR TITLE
22108: Added Pagination components and increased base table text color contrast, MINOR

### PR DESCRIPTION
**Pull request by lancegliser**

### CODE REVIEW

**Comment by lancegliser** [comment]
Might need to add more in support of 22108. Hold off on merging.

**Review by makanadumlao — APPROVED** [review]

**Review by makanadumlao — APPROVED** [review]

File: src/components/Navigation/Pagination/(components)/PaginationBase/PaginationBase.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        className,
+      )}
+    >
+      <PaginationStatus
+        count={count}
+        currentPage={currentPage}
+        loading={loading}
+        pageSize={pageSize}
+      />
+      <div>{controls}</div>
```
[the remark this answers is not in the document]
Comment (lancegliser) [reply src/components/Navigation/Pagination/(components)/PaginationBase/PaginationBase.tsx:39]: Where?

image

image

File: src/components/Navigation/Pagination/PaginationPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+};
+/** A specialized <Button /> with content and styling predefined */
+const Controls: FC<ControlsProps> = ({ count, loading, to, ...props }) => {
+  const { t } = useTranslation(i18n.namespace);
+
+  return (
+    <SecondaryButton
+      as={to ? Link : undefined}
+      to={to}
+      disabled={!count}
```
Comment (fulpm) [inline src/components/Navigation/Pagination/PaginationPreview.tsx:47]: When this is a Link and disabled, it can still be clicked for navigation

**Review by fulpm — APPROVED** [review]

File: src/components/Navigation/Pagination/(components)/PaginationBase/PaginationBase.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lancegliser) [reply src/components/Navigation/Pagination/(components)/PaginationBase/PaginationBase.tsx:39]: Appears to be an implementation issue in consuming systems. The margins on this component are alright.

File: src/components/Navigation/Pagination/PaginationPreview.tsx (hunk 2 above)
Comment (lancegliser) [reply src/components/Navigation/Pagination/PaginationPreview.tsx]: Nice catch `className={twMerge(disabled && "pointer-events-none")}` helps.

**Pull request merged**